src/container/streamAnalyticsManager.ts

```diff
@@ -0,0 +1,108 @@
+                throw new UserCancelledError();
+            }
+        }
+
+        const job: StreamAnalyticsPicItem = await vscode.window.showQuickPick(this.loadAllStreamingJobs(), { placeHolder: "Select Azure Stream Analytics Jobs", ignoreFocusOut: true });
+        return job;
+    }
+
+    public getImageName(): string {
+        return this.STREAM_ANALYTICS_IMAGE;
+    }
+
+    public async getJobInfo(streamingJob: StreamAnalyticsPicItem): Promise<object> {
+        const subscriptionId: string = streamingJob.azureSubscription.subscription.id;
+        const id: string = streamingJob.job.id;
+        const resourceGroup: string = id.slice(id.toLowerCase().search("resourcegroups/") + "resourcegroups/".length, id.toLowerCase().search("/providers/"));
+        const jobName: string = streamingJob.job.name;
+        const publishJobUrl: string = `https://management.azure.com${subscriptionId}/resourceGroups/${resourceGroup}` +
+                                    `/providers/Microsoft.StreamAnalytics/streamingjobs/${jobName}/publishedgepackage?api-version=2017-04-01-preview`;
+
+        const { aadAccessToken } = await this.acquireAadToken(streamingJob.azureSubscription.session);
+
+        const publishResponse = await request.post(publishJobUrl, {
+            auth: {
+                bearer: aadAccessToken,
+            },
+            resolveWithFullResponse: true,
+        });
+
+        const operationResultUrl = publishResponse.headers.location;
+
+        const jobInfoResult = await request.get(operationResultUrl, {
+            auth: {
+                bearer: aadAccessToken,
+            },
+        });
+
+        try {
+            const info = JSON.parse(JSON.parse(jobInfoResult).manifest);
+            return info;
+        } catch (e) {
+            throw new Error("Cannot parse Stream Analytics publish job information!") + e.toString();
+        }
+    }
+
+    private async loadAllStreamingJobs(): Promise<StreamAnalyticsPicItem[]> {
+        try {
+            await this.azureAccount.waitForFilters();
+            const items: StreamAnalyticsPicItem[] = [];
+
+            for (const azureSubscription of this.azureAccount.filters) {
+                const client = new StreamAnalyticsManagementClient(
+                    azureSubscription.session.credentials,
+                    azureSubscription.subscription.subscriptionId!,
+                );
+
+                const streamingJobs = await client.streamingJobs.list();
+                streamingJobs.forEach((job) => {
+                    items.push(new StreamAnalyticsPicItem(job, azureSubscription));
+                });
+            }
+            return items;
+        } catch (error) {
+            error.message = `Error fetching streaming jobs list: ${error.message}`;
+            throw error;
+        }
+    }
+
+    private async acquireAadToken(session: AzureSession): Promise<{ aadAccessToken: string, aadRefreshToken: string }> {
```

Comment:
This method looks duplicate with https://github.com/Microsoft/vscode-azure-iot-edge/blob/master/src/container/acrManager.ts#L159-L174.

Can you extract it and make ASA manager and ACR manager share it? I think https://github.com/Microsoft/vscode-azure-iot-edge/blob/master/src/common/utility.ts may be a candiate place to put common methods like this,

Reply:
Thanks, I have already put it into the utility.ts